<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="LineageDE">
<title>LineageDE-vignette • LineageDE</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="LineageDE-vignette">
<meta property="og:description" content="LineageDE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">LineageDE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/LineageDE-vignette.html">LineageDE-vignette</a>
    <a class="dropdown-item" href="../articles/Pseudotime_Generation_vignette.html">Pseudotime_Generation_vignette</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>LineageDE-vignette</h1>
            
      
      
      <div class="d-none name"><code>LineageDE-vignette.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="example-running-lineage-de">Example running Lineage DE<a class="anchor" aria-label="anchor" href="#example-running-lineage-de"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">LineageDE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-in-example-data">Load in Example Data<a class="anchor" aria-label="anchor" href="#load-in-example-data"></a>
</h3>
<p>This example data already contains the pseudotime results needed to run LineageDE. See Pseudotime_Generation_vignette for how this input is generated The function samplePseudotime can be utilized to generate your own pseudotime results using Slingshot (Street et al. BMC Genomics 2018)</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ExampleInputData"</span><span class="op">)</span></span></code></pre></div>
<p>This input Data contains gene expression information (counts) across HSPCs (hematopoietic stem and progenitor cells) from two conditions (WT &amp; IkBKO) and two replicates (Rep1 &amp; Rep2). (Data subsampled from publication Singh et al.)</p>
<p>ori.pseudotime and ori.weights come from the original pseudotime calculation using all data. sub.pseudotime and sub.weights come from the pseudotime calculations on subsets of sampled data.</p>
<p>Note: to take advantage of parallelization on genes and if RhpcBLASctl installed, run the following command to avoid multithreaded operations:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#RhpcBLASctl::blas_set_num_threads(1)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-1-choose-appropriate-value-for-k">Step 1: Choose appropriate value for k<a class="anchor" aria-label="anchor" href="#step-1-choose-appropriate-value-for-k"></a>
</h3>
<p>We first restrict to only potentially testing genes with nonzero expression in at least 5% of cells (if mostly zero counts, model fitting will fail as k increases).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes2test</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="op">(</span><span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">counts</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span><span class="op">&gt;=</span><span class="op">(</span><span class="fl">0.05</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; Loading required package: Matrix</span></span>
<span><span class="co">#&gt; Warning: package 'Matrix' was built under R version 4.1.3</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genes2test</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 9810</span></span></code></pre></div>
<p>In this example this is performed only for lineage number 1 &amp; sampling only 10 genes to reduce runtime (set parallel = TRUE, n_cores = # to make sampling more genes faster; utilizes BiocParallel)</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">k_results</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/chooseK.html">chooseK</a></span><span class="op">(</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">counts</span>, <span class="va">ExampleInputData</span><span class="op">$</span><span class="va">ori.pseudotimes</span>, <span class="va">ExampleInputData</span><span class="op">$</span><span class="va">conditions</span>, weights<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">ori.weights</span>, experiments<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">experiments</span>, linCompare<span class="op">=</span><span class="fl">1</span>, n<span class="op">=</span><span class="fl">10</span>, genes<span class="op">=</span><span class="va">genes2test</span>, save.name <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Khk"</span></span>
<span><span class="co">#&gt; [1] "Lrpap1"</span></span>
<span><span class="co">#&gt; [1] "Ppp2r5d"</span></span>
<span><span class="co">#&gt; [1] "Cdk8"</span></span>
<span><span class="co">#&gt; [1] "Ube2j1"</span></span>
<span><span class="co">#&gt; [1] "Rcl1"</span></span>
<span><span class="co">#&gt; [1] "Apobec1"</span></span>
<span><span class="co">#&gt; [1] "Panx1"</span></span>
<span><span class="co">#&gt; [1] "Rad51b"</span></span>
<span><span class="co">#&gt; [1] "Plbd2"</span></span>
<span><span class="co">#&gt; [1] "choosing K done!"</span></span>
<span></span>
<span><span class="va">plot_AIC</span> <span class="op">=</span> <span class="va">k_results</span><span class="op">$</span><span class="va">AIC</span><span class="op">/</span><span class="va">k_results</span><span class="op">$</span><span class="va">AIC</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">plot_AIC</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">plot_AIC</span>, <span class="fl">2</span>, <span class="va">mean</span>, na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>, pch<span class="op">=</span><span class="fl">19</span>, ylab<span class="op">=</span><span class="st">"AIC"</span>, xlab<span class="op">=</span><span class="st">"k"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">plot_AIC</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">plot_AIC</span>, <span class="fl">2</span>, <span class="va">mean</span>, na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="LineageDE-vignette_files/figure-html/unnamed-chunk-5-1.png" width="700"> From the above analysis, we choose a value for k from where AIC appears to take a minimum.</p>
</div>
<div class="section level3">
<h3 id="step-2-run-lineagede">Step 2: Run LineageDE<a class="anchor" aria-label="anchor" href="#step-2-run-lineagede"></a>
</h3>
<p>From the prior analysis we choose k = 3. In this example we continue to test lineage 1 &amp; test only 3 genes to reduce runtime (set parallel = TRUE, n_cores = # to make testing more genes faster)</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">k</span><span class="op">=</span><span class="fl">3</span></span>
<span></span>
<span><span class="va">DE_results</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/LineageDE.html">LineageDE</a></span><span class="op">(</span>genes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Hlf"</span>, <span class="st">"Tbp"</span><span class="op">)</span>, <span class="va">ExampleInputData</span><span class="op">$</span><span class="va">counts</span>, <span class="va">ExampleInputData</span><span class="op">$</span><span class="va">ori.pseudotimes</span>, <span class="va">ExampleInputData</span><span class="op">$</span><span class="va">conditions</span>, ori.weights<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">ori.weights</span>, experiments<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">experiments</span>, linCompare<span class="op">=</span><span class="fl">1</span>, k<span class="op">=</span><span class="va">k</span>, samples<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">samples</span>, sub.pseudotimes<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">sub.pseudotimes</span>, sub.weights<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">sub.weights</span>, save.name<span class="op">=</span><span class="st">"None"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "running LineageDE"</span></span>
<span><span class="co">#&gt; [1] "Hlf"</span></span>
<span><span class="co">#&gt; [1] "Tbp"</span></span>
<span><span class="co">#&gt; [1] "finished gene split 1 of 1"</span></span>
<span><span class="co">#&gt; [1] "LineageDE done!"</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">DE_results</span><span class="op">)</span></span>
<span><span class="co">#&gt;   gene              LRT          ori.pVal           emp.pVal          par.pVal</span></span>
<span><span class="co">#&gt; 1  Hlf 145.730205320671                 0 0.0099009900990099                 0</span></span>
<span><span class="co">#&gt; 2  Tbp  5.3684300848538 0.146723190678519  0.297029702970297 0.287755672818664</span></span>
<span><span class="co">#&gt;    pVal.adj</span></span>
<span><span class="co">#&gt; 1 0.0000000</span></span>
<span><span class="co">#&gt; 2 0.2877557</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-3-plotting">Step 3: Plotting<a class="anchor" aria-label="anchor" href="#step-3-plotting"></a>
</h3>
<p>Plot Gene Expression and Model Fit along Lineage</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotGene.html">plotGene</a></span><span class="op">(</span><span class="st">"Hlf"</span>, <span class="va">ExampleInputData</span><span class="op">$</span><span class="va">counts</span>, <span class="va">ExampleInputData</span><span class="op">$</span><span class="va">ori.pseudotimes</span>, <span class="va">ExampleInputData</span><span class="op">$</span><span class="va">conditions</span>, weights <span class="op">=</span> <span class="va">ExampleInputData</span><span class="op">$</span><span class="va">ori.weights</span>, experiments <span class="op">=</span> <span class="va">ExampleInputData</span><span class="op">$</span><span class="va">experiments</span>, linCompare<span class="op">=</span><span class="fl">1</span>, k<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="LineageDE-vignette_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="alternative-uses">Alternative Uses<a class="anchor" aria-label="anchor" href="#alternative-uses"></a>
</h2>
<p>LineageDE can also be used to compare different lineages within the same experiment Here we want to compare lineages 3 &amp; 4 (can compare any number of lineages) to one another rather than compare between the two experimental conditions</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">newData</span> <span class="op">=</span> <span class="fu"><a href="../reference/convertInput.html">convertInput</a></span><span class="op">(</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">counts</span>, <span class="va">ExampleInputData</span><span class="op">$</span><span class="va">ori.pseudotimes</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">4</span><span class="op">)</span>, ori.weights<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">ori.weights</span>, experiments<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">experiments</span>, samples<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">samples</span>, sub.pseudotimes<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">sub.pseudotimes</span>, sub.weights<span class="op">=</span><span class="va">ExampleInputData</span><span class="op">$</span><span class="va">sub.weights</span><span class="op">)</span></span></code></pre></div>
<p>Now newData can be plugged into chooseK, LineageDE, and plotGene as presented above with linCompare=1</p>
<p>“conditions” will carry information about which lineage is which</p>
<p>Note: NA in newData$ori.pseudotimes corresponds to cells with no assignment to the lineages of interest</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Apeksha Singh.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
